### PR TITLE
Add documentation note that getAccessToken() returns cached value

### PR DESCRIPTION
**Pull request by chingor13**
See #143

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java [hunk 1]
```diff
@@ -131,6 +130,11 @@ public boolean hasRequestMetadataOnly() {
     return true;
   }
 
+  /**
+   * Returns the cached access token.
+   *
+   * <p>If not set, you should call {@link #refresh()} to fetch and cache an access token.</p>
+   */
   public final AccessToken getAccessToken() {
```
Comment (JustinBeckwith) [inline oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java]: Wouldn't be better to automatically try obtaining the access token if it hasn't been refreshed yet?

Comment (chingor13) [reply oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java]: A couple of issues mentioned in #143:

* We can't change the method signature to throw an `IOException` without breaking downstream dependencies (some we control, others we may not).
* We could catch and swallow the exceptions which could cause extra, failing requests that are invisible to the user and cannot be acted upon.

Comment (chingor13) [reply oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java]: Technically, this library is pre-1.0 so we could make `getAccessToken()` throw `IOException`.

Comment (chingor13) [reply oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java]: `getAccessToken()` by design is meant to be an accessor to a cached value. It doesn't make sense to try to make it refresh the token and changing the behavior will be unexpected for most users.

We'll document it more clearly that this is only an accessor and introduce a smart refresh request in #163

**Pull request merged**